internal/v1/handler_blueprints.go

```diff
@@ -272,18 +272,158 @@ func (h *Handlers) GetBlueprint(ctx echo.Context, id openapi_types.UUID, params
+					Description: fmt.Sprintf("service %s required as enabled by policy is not present", e),
+				})
+				if fixup {
+					blueprint.Customizations.Services = common.ToPtr(common.FromPtr(blueprint.Customizations.Services))
+					blueprint.Customizations.Services.Enabled = common.ToPtr(append(common.FromPtr(blueprint.Customizations.Services.Enabled), e))
+				}
+			}
+		}
+		for _, m := range services.Masked {
+			if blueprint.Customizations.Services == nil || blueprint.Customizations.Services.Masked == nil || !slices.Contains(*blueprint.Customizations.Services.Masked, m) {
+				lintErrors = append(lintErrors, BlueprintLintItem{
+					Name:        "Compliance",
+					Description: fmt.Sprintf("service %s required as masked by policy is not present", m),
+				})
+				if fixup {
+					blueprint.Customizations.Services = common.ToPtr(common.FromPtr(blueprint.Customizations.Services))
+					blueprint.Customizations.Services.Masked = common.ToPtr(append(common.FromPtr(blueprint.Customizations.Services.Masked), m))
+				}
+			}
+		}
+		for _, d := range services.Disabled {
+			if blueprint.Customizations.Services == nil || blueprint.Customizations.Services.Disabled == nil || !slices.Contains(*blueprint.Customizations.Services.Disabled, d) {
+				lintErrors = append(lintErrors, BlueprintLintItem{
+					Name:        "Compliance",
+					Description: fmt.Sprintf("service %s required as disabled by policy is not present", d),
+				})
+				if fixup {
+					blueprint.Customizations.Services = common.ToPtr(common.FromPtr(blueprint.Customizations.Services))
+					blueprint.Customizations.Services.Disabled = common.ToPtr(append(common.FromPtr(blueprint.Customizations.Services.Disabled), d))
+				}
+			}
+		}
+	}
+	if kernel := bp.Customizations.Kernel; kernel != nil {
+		if kernel.Name != "" && (blueprint.Customizations.Kernel == nil || *blueprint.Customizations.Kernel.Name != kernel.Name) {
+			lintErrors = append(lintErrors, BlueprintLintItem{
+				Name:        "Compliance",
+				Description: fmt.Sprintf("kernel name %s required by policy not set", kernel.Name),
+			})
+			if fixup {
+				blueprint.Customizations.Kernel = common.ToPtr(common.FromPtr(blueprint.Customizations.Kernel))
+				blueprint.Customizations.Kernel.Name = common.ToPtr(kernel.Name)
+			}
+		}
+		kernelcmd := strings.Split(kernel.Append, " ")
+		for _, kcmd := range kernelcmd {
+			if blueprint.Customizations.Kernel == nil || !strings.Contains(*blueprint.Customizations.Kernel.Append, kcmd) {
+				lintErrors = append(lintErrors, BlueprintLintItem{
+					Name:        "Compliance",
+					Description: fmt.Sprintf("kernel command line parameter '%s' required by policy not set", kcmd),
+				})
+			}
+			if fixup {
+				blueprint.Customizations.Kernel = common.ToPtr(common.FromPtr(blueprint.Customizations.Kernel))
+				blueprint.Customizations.Kernel.Append = common.ToPtr(fmt.Sprintf("%s %s", common.FromPtr(blueprint.Customizations.Kernel.Append), kcmd))
+			}
+		}
+	}
+	return lintErrors, nil
+}
```

Comment:
I see a pattern that repeats 4-5 times if I am not mistaken, could be maybe generalized cos I think you are onto something in here and more fixups could be added later. Alternatively, the function could be split into separate subfunctions. Maybe having this in a handler is also not good idea, perhaps acceptable for now and separate package later if it grows.

Reply:
Hm I wonder how you'd generalise this because each separate customisation needs a different treatment. Like other than using reflection and making some guesses.

Agreed with the handler point, I think we should detangle the handlers in general a bit in future. Not sure if this is the right PR for it though.

Reply:
I just saw a pattern and did not look into it deeply. Definitely did not mean to use reflection, no :-) This is fine. Thanks.